tests/validation/tests/v3_api/test_logging_e2e.py

```diff
@@ -0,0 +1,224 @@
+    projects = admin_client.list_project(name="System",
+                                         clusterId=cluster.id).data
+    assert len(projects) == 1
+    project = projects[0]
+    sys_p_client = get_project_client_for_token(project, USER_TOKEN)
+    return sys_p_client
+
+
+def create_cluster_logging(config, json_parsing=False):
+    cluster = namespace["cluster"]
+    admin_client = namespace["admin_client"]
+    name = random_test_name("fluentd")
+    return admin_client.create_cluster_logging(name=name,
+                                               clusterId=cluster.id,
+                                               fluentForwarderConfig=config,
+                                               enableJSONParsing=json_parsing,
+                                               outputFlushInterval=5
+                                               )
+
+
+def delete_cluster_logging(cluster_logging):
+    admin_client = namespace["admin_client"]
+    admin_client.delete(cluster_logging)
+
+
+def create_project_logging(config, json_parsing=False):
+    admin_client = namespace["admin_client"]
+    cluster = namespace["cluster"]
+    projects = admin_client.list_project(name="System",
+                                         clusterId=cluster.id).data
+    assert len(projects) == 1
+    project = projects[0]
+    name = random_test_name("project-fluentd")
+    return admin_client.create_project_logging(name=name,
+                                               projectId=project.id,
+                                               fluentForwarderConfig=config,
+                                               enableJSONParsing=json_parsing,
+                                               outputFlushInterval=5
+                                               )
+
+
+def delete_project_logging(project_logging):
+    admin_client = namespace["admin_client"]
+    admin_client.delete(project_logging)
+
+
+def validate_websocket_view_logs():
+    url_base = 'wss://' + CATTLE_TEST_URL[8:] + \
+               '/k8s/clusters/' + namespace["cluster"].id + \
+               '/api/v1/namespaces/' + namespace["ns"].name + \
+               '/pods/' + namespace["pod"].name + \
+               '/log?container=' + namespace["pod"].containers[0].name
+    params_dict = {
+        "tailLines": 500,
+        "follow": True,
+        "timestamps": True,
+        "previous": False,
+    }
+    params = urllib.parse.urlencode(params_dict, doseq=True,
+                                    quote_via=urllib.parse.quote, safe='()')
+
+    url = url_base + "&" + params
+    wait_for_match(WebsocketLogParse(), url)
+
+
+def wait_for_match(wslog, url, timeout=DEFAULT_TIMEOUT):
+    start = time.time()
+    found = False
+
+    while not found:
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out waiting for string to match in logs")
+        time.sleep(1)
+        ws = create_connection(url, ["base64.binary.k8s.io"])
```

Comment:
We shouldn't be continuously created and closing the connection while waiting for the logs. Open once, check for `last_message` in a loop, and close on a timeout.